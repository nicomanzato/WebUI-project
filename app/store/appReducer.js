'use strict';

import { combineReducers } from 'redux'
import PostReducer from './post/postReducer'
import TrendReducer from './trend/trendReducer'
import UserReducer from './user/userReducer'
import nav from './navigation/navReducer'
import ConfigurationReducer from './configuration/configurationReducer'

const AppReducer = combineReducers({
  PostReducer,
  TrendReducer,
  UserReducer,
  nav,
  ConfigurationReducer,
});

export default AppReducer;
