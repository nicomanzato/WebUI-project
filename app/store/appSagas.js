import { call, all, put, takeEvery, takeLatest, take, select } from 'redux-saga/effects'
import { delay } from 'redux-saga'

import postSagas from './post/postSagas'
import trendSagas from './trend/trendSagas'
import userSagas from './user/userSagas'

export default function* appSagas() {
  yield all([
    postSagas(),
    trendSagas(),
    userSagas(),
  ])
}
