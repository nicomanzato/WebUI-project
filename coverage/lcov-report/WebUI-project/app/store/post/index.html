<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for WebUI-project/app/store/post</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> WebUI-project/app/store/post
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.02% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>173/188</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">97.56% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>40/41</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">97.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>42/43</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.8% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>168/183</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="postActions.js"><a href="postActions.js.html">postActions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="57" class="abs high">57/57</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="57" class="abs high">57/57</td>
	</tr>

<tr>
	<td class="file high" data-value="postReducer.js"><a href="postReducer.js.html">postReducer.js</a></td>
	<td data-value="95.24" class="pic high"><div class="chart"><div class="cover-fill" style="width: 95%;"></div><div class="cover-empty" style="width:5%;"></div></div></td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="42" class="abs high">40/42</td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="21" class="abs high">20/21</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="42" class="abs high">40/42</td>
	</tr>

<tr>
	<td class="file medium" data-value="postSagas.js"><a href="postSagas.js.html">postSagas.js</a></td>
	<td data-value="79.03" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 79%;"></div><div class="cover-empty" style="width:21%;"></div></div></td>
	<td data-value="79.03" class="pct medium">79.03%</td>
	<td data-value="62" class="abs medium">49/62</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="93.75" class="pct high">93.75%</td>
	<td data-value="16" class="abs high">15/16</td>
	<td data-value="79.03" class="pct medium">79.03%</td>
	<td data-value="62" class="abs medium">49/62</td>
	</tr>

<tr>
	<td class="file high" data-value="postSelector.js"><a href="postSelector.js.html">postSelector.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="27" class="abs high">27/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Dec 27 2018 17:02:42 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
