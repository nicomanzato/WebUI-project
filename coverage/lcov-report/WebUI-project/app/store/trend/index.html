<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for WebUI-project/app/store/trend</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> WebUI-project/app/store/trend
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>26/30</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>26/30</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="trendActions.js"><a href="trendActions.js.html">trendActions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="trendReducer.js"><a href="trendReducer.js.html">trendReducer.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="trendSagas.js"><a href="trendSagas.js.html">trendSagas.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Dec 26 2018 13:28:33 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
