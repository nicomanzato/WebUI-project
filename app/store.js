import { applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga'
import appReducer from './store/appReducer'; //Import the reducer
import appSagas from './store/appSagas';
import reactotron from '../ReactotronConfig'

const sagaMonitor = reactotron.createSagaMonitor()


const sagaMiddleware = createSagaMiddleware({ sagaMonitor })

export const store = reactotron.createStore(appReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(appSagas)
