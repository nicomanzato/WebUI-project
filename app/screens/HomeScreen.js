import React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import PropTypes from 'prop-types';
import {getNotSilencedPost} from './../store/post/postSelector'
import {bindActionCreators} from 'redux';
import { connect } from 'react-redux';
import {requestLoadPost, requestLoadMorePost} from '../store/post/postActions';
import Timeline from '../components/timeline/timeline' //Import the component file

export const HomeScreen = (props) => (
  <View style={styles.container}>
    <Timeline
      data={props.loadedPost}
      navigation={props.navigation}
      loading={props.loading}
      loadingMorePost={props.loadingMorePost}
      onRefresh={props.requestLoadPost}
      refreshing={false}
      onComponentMount={props.requestLoadPost}
      onEndReached={props.requestLoadMorePost}
    />
  </View>
);

HomeScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

HomeScreen.navigationOptions = {
  header: null
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFFF'
  },
});

function mapStateToProps(state, props) {
    return {
        loading: state.PostReducer.isLoadingPost,
        loadingMorePost: state.PostReducer.isLoadingMorePost,
        loadedPost: getNotSilencedPost(state),
    }
}

const mapDispatchToProps = {
  requestLoadPost: () => requestLoadPost(),
  requestLoadMorePost: () => requestLoadMorePost(),
}

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
