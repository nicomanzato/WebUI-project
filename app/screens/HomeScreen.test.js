import React from 'react';
import {HomeScreen} from './HomeScreen'
import ReduxWrappedHomeScreen from './HomeScreen'
import { Provider } from 'react-redux'
import PostMock from './../store/post/mock/postMock'
import Timeline from './../components/timeline/timeline'
import { shallow, mount, ReactWrapper } from 'enzyme'
import {store} from './../store'

describe('home screen', () => {

  it('should render correctly', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}}
    const wrapper = shallow(
      <HomeScreen
        navigation={mockNavigationProp}
        loading={false}
        loadingMorePost={false}
        loadedPost={[PostMock]}
        requestLoadPost={() => {}}
        requestLoadMorePost={() => {}}
      />
     );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when wrapped with redux', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}}
    const wrapper = shallow(
    <ReduxWrappedHomeScreen
      store={store}
      navigation={mockNavigationProp}
    />
     );
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle dispatching its actions', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}}
    const requestLoadPost = jest.fn().mockName('mockedFunction');
    const requestLoadMorePost = jest.fn().mockName('mockedFunction');
    const wrapper = mount(
      <HomeScreen
        navigation={mockNavigationProp}
        loading={false}
        loadingMorePost={false}
        loadedPost={[PostMock]}
        requestLoadPost={requestLoadPost}
        requestLoadMorePost={requestLoadMorePost}
      />
    );

    const timeline = wrapper.find(Timeline).first();

    expect(requestLoadPost).toHaveBeenCalled();

    timeline.props().onEndReached();
    expect(requestLoadMorePost).toHaveBeenCalled();
  });

});
