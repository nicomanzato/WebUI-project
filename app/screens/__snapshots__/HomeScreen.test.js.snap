// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home screen should render correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFFF",
      "flex": 1,
    }
  }
>
  <Timeline
    data={
      Array [
        Object {
          "contributors": null,
          "coordinates": null,
          "created_at": "Thu Dec 20 14:06:56 +0000 2018",
          "entities": Object {
            "hashtags": Array [],
            "symbols": Array [],
            "urls": Array [
              Object {
                "display_url": "twitter.com/i/web/status/1…",
                "expanded_url": "https://twitter.com/i/web/status/1075754427368333313",
                "indices": Array [
                  117,
                  140,
                ],
                "url": "https://t.co/WG9oBkfoFF",
              },
            ],
            "user_mentions": Array [],
          },
          "favorite_count": 0,
          "favorited": false,
          "geo": null,
          "id": 1075754427368333300,
          "id_str": "1075754427368333313",
          "in_reply_to_screen_name": null,
          "in_reply_to_status_id": null,
          "in_reply_to_status_id_str": null,
          "in_reply_to_user_id": null,
          "in_reply_to_user_id_str": null,
          "is_quote_status": false,
          "lang": "es",
          "place": null,
          "possibly_sensitive": false,
          "possibly_sensitive_appealable": false,
          "retweet_count": 0,
          "retweeted": false,
          "source": "<a href=\\"http://twitter.com\\" rel=\\"nofollow\\">Twitter Web Client</a>",
          "text": "Por una medida de Trump, Rusia advirtió que \\"hay peligro\\" de una guerra nuclear: \\"Puede morir toda la civilización\\"… https://t.co/WG9oBkfoFF",
          "truncated": true,
          "user": Object {
            "contributors_enabled": false,
            "created_at": "Mon Mar 23 12:35:07 +0000 2009",
            "default_profile": true,
            "default_profile_image": false,
            "description": "Noticias las 24 horas, por el canal líder en audiencia y credibilidad de Argentina",
            "entities": Object {
              "description": Object {
                "urls": Array [],
              },
              "url": Object {
                "urls": Array [
                  Object {
                    "display_url": "tn.com.ar",
                    "expanded_url": "http://www.tn.com.ar",
                    "indices": Array [
                      0,
                      22,
                    ],
                    "url": "http://t.co/TX6TF1nOUs",
                  },
                ],
              },
            },
            "favourites_count": 1,
            "follow_request_sent": false,
            "followers_count": 4677787,
            "following": false,
            "friends_count": 124,
            "geo_enabled": false,
            "has_extended_profile": false,
            "id": 25992212,
            "id_str": "25992212",
            "is_translation_enabled": true,
            "is_translator": false,
            "lang": "es",
            "listed_count": 9085,
            "location": "Argentina",
            "name": "TN - Todo Noticias",
            "notifications": false,
            "profile_background_color": "666666",
            "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_tile": true,
            "profile_banner_url": "https://pbs.twimg.com/profile_banners/25992212/1533675118",
            "profile_image_url": "http://pbs.twimg.com/profile_images/962300034661396481/OzHK6TZr_normal.jpg",
            "profile_image_url_https": "https://pbs.twimg.com/profile_images/962300034661396481/OzHK6TZr_normal.jpg",
            "profile_link_color": "D51D2E",
            "profile_sidebar_border_color": "FFFFFF",
            "profile_sidebar_fill_color": "314BB9",
            "profile_text_color": "000000",
            "profile_use_background_image": true,
            "protected": false,
            "screen_name": "todonoticias",
            "statuses_count": 84280,
            "time_zone": null,
            "translator_type": "none",
            "url": "http://t.co/TX6TF1nOUs",
            "utc_offset": null,
            "verified": true,
          },
        },
      ]
    }
    loading={false}
    loadingMorePost={false}
    navigation={
      Object {
        "state": Object {
          "params": Object {
            "data": "1",
          },
        },
      }
    }
    onComponentMount={[Function]}
    onEndReached={[Function]}
    onRefresh={[Function]}
    refreshing={false}
  />
</Component>
`;

exports[`home screen should render correctly when wrapped with redux 1`] = `
<HomeScreen
  loadedPost={Array []}
  loading={false}
  loadingMorePost={false}
  navigation={
    Object {
      "state": Object {
        "params": Object {
          "data": "1",
        },
      },
    }
  }
  requestLoadMorePost={[Function]}
  requestLoadPost={[Function]}
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
